build-tools/packages/build-cli/src/commands/check/latestVersions.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Args } from "@oclif/core";
+import * as semver from "semver";
+import { findPackageOrReleaseGroup, packageOrReleaseGroupArg } from "../../args.js";
+import { BaseCommand, sortVersions } from "../../library/index.js";
+
+export default class LatestVersionsCommand extends BaseCommand<typeof LatestVersionsCommand> {
+	static readonly summary =
+		"Determines if an input version matches a latest minor release version.";
+
+	static readonly description =
+		"This command is used in CI to determine if a pipeline was triggered by a release branch with the latest minor version of a major version.";
+
+	static readonly args = {
+		version: Args.string({
+			required: true,
+			description: "The version corresponding to the pipeline trigger branch.",
+		}),
+		package_or_release_group: packageOrReleaseGroupArg(),
+	} as const;
+
+	public async run(): Promise<boolean> {
+		const { args } = this;
+		const context = await this.getContext();
+		const versionInput = this.args.version;
+
+		if (args.package_or_release_group === undefined) {
+			this.error("No dependency provided.");
+		}
+
+		const rgOrPackage = findPackageOrReleaseGroup(args.package_or_release_group, context);
+		if (rgOrPackage === undefined) {
+			this.error(`Package not found: ${args.package_or_release_group}`);
+		}
+
+		const versions = await context.getAllVersions(rgOrPackage.name);
+
+		if (!versions) {
+			this.error(`No versions found for ${rgOrPackage.name}`);
+		}
+
+		// Filter out pre-releases and versions with metadata
+		const stableVersions = versions.filter((v) => {
+			return semver.valid(v.version) !== null && semver.prerelease(v.version) === null;
```

Comment:
Any reason to believe the values returned by getAllVersions wouldn't be valid? That check may be unnecessary.

Reply:
Also, you shouldn't get any "prerelease" versions since we're using the git tags. If your goal is to remove internal versions (e.g. 2.0.0-internal.8.0.0) then use isInternalVersion from version-tools.